data-finding/src/PricePredictor.py

```diff
@@ -1,163 +1,97 @@
+  # cols = ["postal_code", "location_name", "active_listing_count", "median_days_on_market","median_sqft","average_price","total_listing_count","quality_flag"]
+  cols = [i for i in args]
+  # logging.debug(cols)
   df = df.drop(cols, axis=1)
-  df.info()
+  # df.info() #Extra info for debugging
 
   #Either drop or create normalized values for null values
   #Dropping because median listing price is important
   df = df.dropna()
-  df.info()
-
-  from sklearn.model_selection import train_test_split
+  # df.info() #Extra info for debugging
 
-  #transfer to usable data
 
 
-  df.date = pd.to_datetime(df.date)
-  x = df.date.values
-  # x_temp = dateTransfer(x)
-  y = df['median_price'].values
-  print(x)
-  print(y)
-  # x = np.delete(x, 1, axis = 1)
+  #transfer to usable data
+  x = df[kwargs['x_axis']].values
+  y = df[kwargs['y_axis']].values
+  # logging.info(x)
+  # logging.info(y)
 
   x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.3, random_state=0)
 
-  plot.plot(x_train, y_train, 'ro')
-  plot.title("Train data")
-  plot.show()
+  # Extra data for debugging
+  # plot.plot(x_train, y_train, 'ro')
+  # plot.title("Train data")
+  # plot.show()
 
-  print(x_train)
-  print(y_train)
+  # logging.info(x_train)
+  # logging.info(y_train)
 
-  plot.plot(x_test, y_test, 'ro')
-  plot.title("Test data")
-  plot.show()
-  print(x_test)
-  print(y_test)
+  # Extra data for debugging
+  # plot.plot(x_test, y_test, 'ro')
+  # plot.title("Test data")
+  # plot.show()
+
+  # logging.info(x_test)
+  # logging.info(y_test)
 
-  from sklearn.linear_model import LinearRegression
 
   reg = LinearRegression()# Create the model
   x_train = x_train.reshape(-1,1) # Reshape the data to a 2D array as it is currently a 1D array
   y_train = y_train.reshape(-1,1)
   reg.fit(x_train, y_train)#Fit the model and use training data
 
-  y_pred = reg.predict(df.date.values.astype(float).reshape(-1,1)) #Predict off the entire dataset and add prediction to the dataframe
+  y_pred = reg.predict(df[kwargs['x_axis']].values.astype(float).reshape(-1,1)) #Predict off the entire dataset and add prediction to the dataframe
   df['pred'] = y_pred
 
-  ax = df.plot(x='date', y='median_price', color='black', style='.')#Plot the data
-  df.plot(x='date', y='pred', color='orange', linewidth=3, ax=ax, alpha = 0.5)
-  ax.set_title("Prediction of Zip code 77433")
-  ax.set_xlabel("Date")
-  ax.set_ylabel("Median Price")
+  ax = df.plot(x=kwargs['x_axis'], y=kwargs['y_axis'], color='black', style='.')#Plot the data
+  df.plot(x=kwargs['x_axis'], y='pred', color='orange', linewidth=3, ax=ax, alpha = 0.5)
+  ax.set_title("Prediction of zip code")
+  ax.set_xlabel(kwargs['x_axis'])
+  ax.set_ylabel(kwargs['y_axis'])
+  pred_score = reg.score(df[kwargs['x_axis']].values.astype(float).reshape(-1,1),df[kwargs['y_axis']].values)
+  return [pred_score, ax]
+
+
+# graphXY("77433data.csv", 0, 4)
+
+# Example use of linear regression
+if __name__ == "__main__":
```

Comment:
https://docs.python.org/3/library/__main__.html

You should not have the if check for main here, that's only for the entrypoint, looks like that's pricepredictory.py